.github/workflows/ci.yml

```
@@ -498,5 +498,10 @@ jobs:
     # https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4?u=graingert
     runs-on: ubuntu-latest
     needs: [linux, linux-distros, linux-rust, linux-rust-coverage, macos, windows, linux-downstream]
+    if: ${{ always() }}
     steps:
+    - run: echo "😢";exit 1
+      if: ${{ needs.linux.result == 'failure' || needs.linux-distros.result == 'failure' || needs.linux-rust.result == 'failure' || needs.linux-rust-coverage.result == 'failure' || needs.macos.result == 'failure' || needs.windows.result == 'failure' || needs.linux-downstream.result == 'failure' }}
+      # this step will run if the previous step is skipped because there is an implicit
+      # if: ${{ success() }} on all steps
```

```suggestion
    # this step will run if the previous step is skipped because there is an implicit
    # if: ${{ success() }} on all steps
```

or delete, not sure how useful this comment is